src/nft/components/profile/list/utils.ts

```diff
@@ -123,69 +108,89 @@ export const getListings = (sellAssets: WalletAsset[]): [CollectionRow[], Listin
   return [newCollectionsToApprove, newListings]
 }
 
-type ListingState = {
-  allListingsPending: boolean
-  allListingsDefined: boolean
-  allListingsApproved: boolean
-  allCollectionsPending: boolean
-  allCollectionsDefined: boolean
-  anyActiveSigning: boolean
-  anyActiveFailures: boolean
-  anyActiveRejections: boolean
-  anyPaused: boolean
+export const verifyStatus = (status: ListingStatus) => {
+  return status !== ListingStatus.PAUSED && status !== ListingStatus.APPROVED
 }
 
-export const getListingState = (
-  collectionsRequiringApproval: CollectionRow[],
-  listings: ListingRow[]
-): ListingState => {
-  let allListingsPending = true
-  let allListingsDefined = true
-  let allListingsApproved = true
-  let allCollectionsPending = true
-  let allCollectionsDefined = true
-  let anyActiveSigning = false
-  let anyActiveFailures = false
-  let anyActiveRejections = false
-  let anyPaused = false
+export function useSubscribeListingState() {
+  const sellAssets = useSellAsset((state) => state.sellAssets)
+  const { setListings, setCollectionsRequiringApproval } = useNFTList(
+    ({ setListings, setCollectionsRequiringApproval }) => ({
+      setListings,
+      setCollectionsRequiringApproval,
+    }),
+    shallow
+  )
+  useEffect(() => {
+    const [newCollectionsToApprove, newListings] = getListings(sellAssets)
+    setListings(newListings)
+    setCollectionsRequiringApproval(newCollectionsToApprove)
+  }, [sellAssets, setCollectionsRequiringApproval, setListings])
+}
 
-  if (collectionsRequiringApproval.length === 0) {
-    allCollectionsDefined = allCollectionsPending = false
-  }
-  for (const collection of collectionsRequiringApproval) {
-    if (collection.status !== ListingStatus.PENDING) allCollectionsPending = false
-    if (collection.status !== ListingStatus.DEFINED) allCollectionsDefined = false
-    if (collection.status === ListingStatus.SIGNING) anyActiveSigning = true
-    else if (collection.status === ListingStatus.FAILED) anyActiveFailures = true
-    else if (collection.status === ListingStatus.REJECTED) anyActiveRejections = true
-    else if (collection.status === ListingStatus.PAUSED) anyPaused = true
-  }
+export function useHandleGlobalPriceToggle(
+  globalOverride: boolean,
+  setListPrice: Dispatch<number | undefined>,
+  setPrice: (price?: number) => void,
+  listPrice?: number,
+  globalPrice?: number
+) {
+  useEffect(() => {
+    let price: number | undefined
+    if (globalOverride) {
+      if (!listPrice) setListPrice(globalPrice)
+      price = globalPrice
+    } else {
+      price = listPrice
+    }
+    setPrice(price)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
do we have to disable eslint here? I don't see anything we can't add to the deps

Reply:
Same as below, we want this effect to be atomic to only when a user toggles the button to link/unlink an input row from the Same Price method, otherwise it unintentionally will write into the inputs